internal/commands/build/build.go

```diff
@@ -0,0 +1,240 @@
+package build
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/deislabs/cnab-go/bundle"
+	cnab "github.com/deislabs/cnab-go/driver"
+	"github.com/docker/app/internal/packager"
+	"github.com/docker/app/types"
+	"github.com/docker/buildx/build"
+	"github.com/docker/buildx/driver"
+	_ "github.com/docker/buildx/driver/docker" // required to get default driver registered, see driver/docker/factory.go:14
+	"github.com/docker/buildx/util/progress"
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/distribution/reference"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/auth/authprovider"
+	"github.com/moby/buildkit/util/appcontext"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type buildOptions struct {
+	noCache  bool
+	progress string
+	pull     bool
+	tag      string
+	out      string
+}
+
+func Cmd(dockerCli command.Cli) *cobra.Command {
+	var opts buildOptions
+	cmd := &cobra.Command{
+		Use:     "build [APPLICATION]",
+		Short:   "Build service images for the application",
+		Example: `$ docker app build myapp.dockerapp`,
+		Args:    cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			tag, err := runBuild(dockerCli, args[0], opts)
+			if err == nil {
+				fmt.Printf("Successfully build %s\n", tag.String())
+			}
+			return err
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.noCache, "no-cache", false, "Do not use cache when building the image")
+	flags.StringVar(&opts.progress, "progress", "auto", "Set type of progress output (auto, plain, tty). Use plain to show container output")
+	flags.BoolVar(&opts.pull, "pull", false, "Always attempt to pull a newer version of the image")
+	flags.StringVarP(&opts.out, "output", "o", "", "Dump generated bundle into a file")
+	flags.StringVarP(&opts.tag, "tag", "t", "", "Name and optionally a tag in the 'name:tag' format")
+
+	return cmd
+}
+
+func runBuild(dockerCli command.Cli, application string, opt buildOptions) (reference.Named, error) {
+	err := checkMinimalEngineVersion(dockerCli)
+	if err != nil {
+		return nil, err
+	}
+
+	var ref reference.Named
+	ref, err = packager.GetNamedTagged(opt.tag)
+	if err != nil {
+		return nil, err
+	}
+
+	app, err := packager.Extract(application)
+	if err != nil {
+		return nil, err
+	}
+	defer app.Cleanup()
+
+	bundle, err := packager.MakeBundleFromApp(dockerCli, app, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	buildopts, err := parseCompose(app, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	buildopts["invocation-image"], err = createInvocationImageBuildOptions(dockerCli, app)
+	if err != nil {
+		return nil, err
+	}
+
+	debugBuildOpts(buildopts)
+
+	ctx, cancel := context.WithCancel(appcontext.Context())
+	defer cancel()
+	d, err := driver.GetDriver(ctx, "buildx_buildkit_default", nil, dockerCli.Client(), nil, "", nil)
```

Comment:
Can we extract this `buildx_buildkit_default` constant?

Reply:
afaik it doesn't have any specific meaning, only for those actively using buildx who will track the active drivers.

Reply:
Yep, when I mean "extract" I just mean:
```go
// buildxDefaultDriver is the default driver used by buildx for buildkit blablabla
const buildxDefaultDriver = "buildx_buildkit_default"
d, err := driver.GetDriver(ctx, buildxDefaultDriver, nil, dockerCli.Client(), nil, "", nil)
```